### DISCUSSION TITLE
any suggest for easist way to customized YARD for my scenario as below?

### DISCUSSION THREAD

**Discussion by rainmanms** [dotnet/yarp#2573] [Q&A] [answered]
i played the simple version of proxy, it works well... so i wanted to do some customize to fulfill my own requirements. what i need are:
1. when one desitination is not working well, for example, the latency is higher than pre-define threshold, then we would take that destination out for 1 hours for example. later if try and it works, put it back to destination list;
2. if some call to destination failed with any reason, we would like to update some input paramter (actually update the payload in http request), and then sent it to destination again. 
3. customized the response header with desitination infromation, like name or url

any pointer what is the easist way to make it with YARD lib would be really apperiacted.

**Comment by MihaZupan** [comment, marked as answer]
> when one desitination is not working well, for example, the latency is higher than pre-define threshold, then we would take that destination out for 1 hours for example. later if try and it works, put it back to destination list;

The built-in health checks don't treat slow destinations (high latency) as problematic. You could look into implementing a custom health check policy. Either active or passive depending on your needs.
See https://microsoft.github.io/reverse-proxy/articles/dests-health-checks.html

> if some call to destination failed with any reason, we would like to update some input paramter (actually update the payload in http request), and then sent it to destination again.

Support for retries isn't built-in yet, but you can look at the discussion in https://github.com/microsoft/reverse-proxy/issues/56 for how to go about implementing something like that.

> customized the response header with desitination infromation, like name or url

See https://microsoft.github.io/reverse-proxy/articles/transforms.html

E.g.
```c#
.AddTransforms(context =>
{
    context.AddResponseTransform(context =>
    {
        context.HttpContext.Response.Headers["X-Custom-Header"] = context.HttpContext.GetReverseProxyFeature().ProxiedDestination?.DestinationId;
        return default;
    });
});
```
